Index: src/unix/Makefile.in
===================================================================
--- src/unix/Makefile.in	(revision 45534)
+++ src/unix/Makefile.in	(revision 45535)
@@ -78,11 +78,11 @@
 	  -e 's/@VERSION/$(VERSION)/' $(srcdir)/libR.pc.in \
 	  > "$(DESTDIR)$(libdir)/pkgconfig/libR.pc"
 
-install: install-main install-Rscript
+install: install-main
 install: @WANT_R_SHLIB_TRUE@ install-pc
 install: @WANT_R_STATIC_TRUE@ install-pc
 
-install-main:
+install-main: install-Rscript
 	@$(MKINSTALLDIRS) "$(DESTDIR)$(bindir)"
 	@$(INSTALL_PROGRAM) Rscript "$(DESTDIR)$(bindir)/Rscript"
 
